src/utils/__tests__/cosmosdb_model.test.ts

```diff
@@ -0,0 +1,239 @@
+import * as t from "io-ts";
+
+import { isLeft, isRight } from "fp-ts/lib/Either";
+
+import { Container, ResourceResponse } from "@azure/cosmos";
+
+import { fromNullable, isNone } from "fp-ts/lib/Option";
+import { BaseModel, CosmosdbModel, ResourceT } from "../cosmosdb_model";
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+const MyDocument = t.interface({
+  test: t.string
+});
+type MyDocument = t.TypeOf<typeof MyDocument>;
+
+const NewMyDocument = t.intersection([MyDocument, BaseModel]);
+type NewMyDocument = t.TypeOf<typeof NewMyDocument>;
+
+const RetrievedMyDocument = t.intersection([MyDocument, ResourceT]);
+type RetrievedMyDocument = t.TypeOf<typeof RetrievedMyDocument>;
+
+class MyModel extends CosmosdbModel<
+  MyDocument,
+  NewMyDocument,
+  RetrievedMyDocument
+> {
+  constructor(c: Container) {
+    super(c, NewMyDocument, RetrievedMyDocument);
+  }
+}
+
+const readMock = jest.fn();
+const containerMock = {
+  item: jest.fn(),
+  items: {
+    create: jest.fn(),
+    upsert: jest.fn()
+  }
+};
+const container = (containerMock as unknown) as Container;
+
+const aDocument = {
+  id: "test-id-1",
+  test: "test"
+};
+
+export const someMetadata = {
+  _etag: "_etag",
+  _rid: "_rid",
+  _self: "_self",
+  _ts: 1
+};
+
+describe("create", () => {
+  it("should create a document", async () => {
+    containerMock.items.create.mockResolvedValueOnce(
+      new ResourceResponse(
+        {
+          ...aDocument,
+          ...someMetadata
+        },
+        {},
+        200,
+        200
+      )
+    );
+    const model = new MyModel(container);
+    const result = await model.create(aDocument).run();
+    expect(containerMock.items.create).toHaveBeenCalledWith(aDocument, {
+      disableAutomaticIdGeneration: true
+    });
+    expect(isRight(result));
+    if (isRight(result)) {
+      expect(result.value).toEqual({
+        ...aDocument,
+        ...someMetadata
+      });
+    }
+  });
+
+  it("should fail on query error", async () => {
+    containerMock.items.create.mockRejectedValueOnce({ statusCode: 500 });
+    const model = new MyModel(container);
+
+    const result = await model.create(aDocument).run();
+    expect(isLeft(result));
+    if (isLeft(result)) {
+      expect(result.value.kind).toBe("COSMOS_ERROR_RESPONSE");
+      if (
+        result.value.kind === "COSMOS_ERROR_RESPONSE" &&
+        fromNullable(result.value.error.code).isSome()
```

Comment:
@AleDore this branch gets never executed as `code` is always `undefined`, `create` should return an `ErrorResponse` that has `code: 500`

Reply:
It depends from previous wrong mock, but I've noticed that `error.code` is always undefined, even if you mock it with `{code: 500}`

Reply:
Yes the mock doesn't work, the method toCosmosErrors doesn't recognize it as an ErrorRrsponse, the mock should return an ErrorResponse